Preview button crashes language change due to change in number of React hooks

If we have the button on a piece of content, and then switch language, the page goes blank and the following error shows up in the console:

```
main.9296bcc6.js:10278 Error: Minified React error #310; visit https://reactjs.org/docs/error-decoder.html?invariant=310 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Ti (main.9296bcc6.js:10276:55391)
    at Object.Ka [as useRef] (main.9296bcc6.js:10276:58831)
    at p.useRef (main.9296bcc6.js:10307:5860)
    at $.default (main.9296bcc6.js:9880:5574)
    at h (main.9296bcc6.js:10224:16156)
    at M (main.9296bcc6.js:10224:16852)
    at Zu (main.9296bcc6.js:10276:54833)
    at Ds (main.9296bcc6.js:10276:64692)
    at Fc (main.9296bcc6.js:10280:10162)
    at Rc (main.9296bcc6.js:10280:992)
```

If you follow the link, you'll discover that the issue is due to "Rendered more hooks than during the previous render."

Skimming the code, a candidate for this problem could be here: https://github.com/mattmilburn/strapi-plugin-preview-button/blob/main/admin/src/components/Injector/index.js

Namely:
```js
  if ( ! shouldRender ) {
    return null;
  }

  const { data, isLoading } = usePreviewData( uid, id, [ initialData ] );
```

In this case, there's a risk that `null` will be returned, which means that `usePreviewData` is never called. React (for good reason) insists that hooks should never be conditionally run, hence the error.

One solution could be to have the hook return a load function instead of a result. Another is to introduce a `skip` variable like e.g. Apollo does in their `useQuery`, meaning that the hook is always called but could just do nothing if need be.

By the way, reloading the blank page makes it work again until you change language once again.

Looking forward to this plugin working with i18n content. :)

--- a/admin/src/components/Injector/index.js
+++ b/admin/src/components/Injector/index.js
@@ -1,10 +1,8 @@
 import React from 'react';
 import { useParams } from 'react-router-dom';
-import { useSelector } from 'react-redux';
 import { useCMEditViewDataManager } from '@strapi/helper-plugin';
 
 import { usePreviewData } from '../../hooks';
-import { pluginId } from '../../utils';
 import { PreviewButton } from '../';
 
 const Injector = () => {
@@ -17,18 +15,13 @@ const Injector = () => {
   } = useCMEditViewDataManager();
   const { id } = useParams();
   const { uid } = allLayoutData.contentType;
-  const { contentTypes } = useSelector( state => state[ `${pluginId}_config` ].config );
+  const {
+    data,
+    isLoading,
+    isSupportedType,
+  } = usePreviewData( uid, id, isCreatingEntry, [ initialData ] );
 
-  const isSupportedType = contentTypes && contentTypes.includes( uid );
-  const shouldRender = isSupportedType && ! isCreatingEntry;
-
-  if ( ! shouldRender ) {
-    return null;
-  }
-
-  const { data, isLoading } = usePreviewData( uid, id, [ initialData ] );
-
-  if ( isLoading || ! data || ! data?.urls ) {
+  if ( ! isSupportedType || isCreatingEntry || isLoading || ! data || ! data?.urls ) {
     return null;
   }
 

--- a/admin/src/hooks/use-preview-data.js
+++ b/admin/src/hooks/use-preview-data.js
@@ -27,19 +27,30 @@ const fetchData = async ( uid, id, toggleNotification ) => {
   }
 };
 
-const usePreviewData = ( uid, id, fetchDependencies ) => {
+const usePreviewData = ( uid, id, isCreatingEntry, fetchDependencies = [] ) => {
   const dispatch = useDispatch();
   const toggleNotification = useNotification();
   const data = useSelector( state => state[ pluginId ].data );
   const isLoading = useSelector( state => state[ pluginId ].isLoading );
+  const { contentTypes } = useSelector( state => state[ `${pluginId}_config` ].config );
+
+  const isSupportedType = contentTypes && contentTypes.includes( uid );
 
   useEffect( () => {
+    if ( ! isSupportedType || isCreatingEntry ) {
+      return;
+    }
+
     fetchData( uid, id, toggleNotification ).then( data => {
       dispatch( { type: RESOLVE_PREVIEW, data } );
     } );
   }, [ dispatch, toggleNotification, ...fetchDependencies ] );
 
-  return { data, isLoading };
+  return {
+    data,
+    isLoading,
+    isSupportedType,
+  };
 };
 
 export default usePreviewData;

